activesupport/lib/active_support/core_ext/range/compare_range.rb

```diff
@@ -36,21 +24,44 @@ def ===(value)
     # The native Range#include? behavior is untouched.
     #  ('a'..'f').include?('c') # => true
     #  (5..9).include?(11) # => false
-    #
-    # The given range must be fully bounded, with both start and end.
-    def include?(value)
-      if value.is_a?(::Range)
-        is_backwards_op = value.exclude_end? ? :>= : :>
-        return false if value.begin && value.end && value.begin.public_send(is_backwards_op, value.end)
+    def include?(other)
+      compare_with_range(other) { |arg_for_comparison| super(arg_for_comparison) }
+    end
+
+    private
+      def compare_with_range(other, &block)
+        if other.is_a?(::Range)
```

Comment:
let's move this condition outside. `compare_with_range` should only ever be called if `other` is indeed a range. also we could make `include?` an alias of `===`. they're essentially the same. in the end we should have something like this:

```ruby
def ===(other)
  return compare_with_range(other) if other.is_a?(::Range)

  super(other)
end

alias :include? :===
```